src/router/routes/dashboard.tsx

```diff
@@ -19,76 +20,81 @@ const OrganizationTeam = lazy(() => import('~elements/dashboard/team'))
+  const queryClient = useQueryClient()
+  const { client, account } = useClient()
+
+  return {
     element: (
       <SuspenseLoader>
-        <LayoutDashboard />
+        <OrganizationProtectedRoute />
       </SuspenseLoader>
     ),
     children: [
       {
-        path: Routes.dashboard.base,
-        element: (
-          <SuspenseLoader>
-            <OrganizationDashboard />
-          </SuspenseLoader>
-        ),
-      },
-      {
-        path: Routes.dashboard.process,
-        element: (
-          <SuspenseLoader>
-            <DashboardProcessView />
-          </SuspenseLoader>
-        ),
-        loader: async ({ params }: { params: Params<string> }) => client.fetchElection(params.id),
-        errorElement: <Error />,
-      },
-      {
-        path: Routes.dashboard.organization,
-        element: (
-          <SuspenseLoader>
-            <OrganizationEdit />
-          </SuspenseLoader>
-        ),
-      },
-      {
-        path: Routes.dashboard.profile,
         element: (
           <SuspenseLoader>
-            <Profile />
-          </SuspenseLoader>
-        ),
-      },
-      {
-        path: Routes.dashboard.processes,
-        element: (
-          <SuspenseLoader>
-            <DashboardProcesses />
-          </SuspenseLoader>
-        ),
-      },
-      {
-        path: Routes.dashboard.team,
-        element: (
-          <SuspenseLoader>
-            <OrganizationTeam />
+            <LayoutDashboard />
           </SuspenseLoader>
         ),
+        children: [
+          {
+            path: Routes.dashboard.base,
+            element: (
+              <SuspenseLoader>
+                <OrganizationDashboard />
+              </SuspenseLoader>
+            ),
+          },
+          {
+            path: Routes.dashboard.process,
+            element: (
+              <SuspenseLoader>
+                <DashboardProcessView />
+              </SuspenseLoader>
+            ),
+            loader: async ({ params }: { params: Params<string> }) => client.fetchElection(params.id),
+            errorElement: <Error />,
+          },
+          {
+            path: Routes.dashboard.organization,
+            element: (
+              <SuspenseLoader>
+                <OrganizationEdit />
+              </SuspenseLoader>
+            ),
+          },
+          {
+            path: Routes.dashboard.profile,
+            element: (
+              <SuspenseLoader>
+                <Profile />
+              </SuspenseLoader>
+            ),
+          },
+          {
+            path: Routes.dashboard.processes,
+            element: (
+              <SuspenseLoader>
+                <DashboardProcesses />
+              </SuspenseLoader>
+            ),
+            loader: async ({ params }) => {
+              const query = paginatedElectionsQuery(account, client, params)
+              return queryClient.getQueryData(query.queryKey) ?? (await queryClient.fetchQuery(query))
```

Comment:
This can be refactored to:

```ts
return queryClient.ensureQueryData(query)
```

Read more:

https://tkdodo.eu/blog/react-query-meets-react-router#getquerydata--fetchquery
https://tanstack.com/query/v4/docs/reference/QueryClient#queryclientensurequerydata

Reply:
Thanks, changed 728455dd4b5d02fe1b1202534efc866562022ffe